nixos/modules/programs/neovim.nix

```diff
@@ -0,0 +1,165 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.neovim;
+
+  runtime' = filter (f: f.enable) (attrValues cfg.runtime);
+
+  # taken from the etc module
+  runtime = pkgs.stdenvNoCC.mkDerivation {
+    name = "runtime";
+
+    builder = ../system/etc/make-etc.sh;
+
+    preferLocalBuild = true;
+    allowSubstitutes = false;
+
+    sources = map (x: x.source) runtime';
+    targets = map (x: x.target) runtime';
+  };
+
+in {
+  options.programs.neovim = {
+    enable = mkEnableOption "Neovim";
+
+    defaultEditor = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        When enabled, installs neovim and configures neovim to be the default editor
+        using the EDITOR environment variable.
+      '';
+    };
+
+    viAlias = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Symlink <command>vi</command> to <command>nvim</command> binary.
+      '';
+    };
+
+    vimAlias = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Symlink <command>vim</command> to <command>nvim</command> binary.
+      '';
+    };
+
+    withRuby = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Enable ruby provider.";
+    };
+
+    configure = mkOption {
+      type = types.attrs;
+      default = {};
+      example = literalExample ''
+        configure = {
+            customRC = $''''
+            " here your custom configuration goes!
+            $'''';
+            packages.myVimPackage = with pkgs.vimPlugins; {
+              # loaded on launch
+              start = [ fugitive ];
+              # manually loadable by calling `:packadd $plugin-name`
+              opt = [ ];
+            };
+          };
+      '';
+      description = ''
+        Generate your init file from your list of plugins and custom commands.
+        Neovim will then be wrapped to load <command>nvim -u /nix/store/<hash>-vimrc</command>
```

Comment:
```
building '/nix/store/f49frvgmnk7j528awy3z45p36rc3rhcl-options-docbook.xml.drv'...
intermediate.xml:8247: parser error : Opening and ending tag mismatch: hash line 8247 and command
m will then be wrapped to load <command>nvim -u /nix/store/<hash>-vimrc</command
                                                                               ^
intermediate.xml:8248: parser error : Opening and ending tag mismatch: command line 8247 and para
</para></nixos:option-description><para><emphasis>Type:</emphasis> attribute set
       ^
intermediate.xml:8248: parser error : Opening and ending tag mismatch: para line 8247 and nixos:option-description
</para></nixos:option-description><para><emphasis>Type:</emphasis> attribute set
                                  ^
intermediate.xml:8267: parser error : Opening and ending tag mismatch: option-description line 8247 and listitem
            </filename></member></simplelist></listitem></varlistentry><varliste
                                                        ^
intermediate.xml:8267: parser error : Opening and ending tag mismatch: listitem line 8247 and varlistentry
            </filename></member></simplelist></listitem></varlistentry><varliste
                                                                       ^
intermediate.xml:75124: parser error : Opening and ending tag mismatch: varlistentry line 8247 and variablelist
      </filename></member></simplelist></listitem></varlistentry></variablelist>
                                                                               ^
intermediate.xml:75124: parser error : Opening and ending tag mismatch: variablelist line 8247 and appendix
lename></member></simplelist></listitem></varlistentry></variablelist></appendix
                                                                               ^
intermediate.xml:75126: parser error : EndTag: '</' not found

^
unable to parse intermediate.xml
```

This will probably block the `unstable-small` channel.

Reply:
Nevermind, it was already reverted.

Reply:
In a fork… github's UI is complicated.

Reply:
Is there a fork for the revert/fix of this?

Reply:
https://github.com/eadwu/nixpkgs/commit/6ecc320b56eb448151b88513ae4a8bc2bee17f8f

Reply:
I will commit a fix

Reply:
here https://github.com/NixOS/nixpkgs/pull/99023

Reply:
@ajs124 does that fix it for you ? I dont have the resorces to build on master but it looked fine.